test/geometry/warp/test_homography_warper.py

```diff
@@ -266,6 +265,15 @@ def test_transform2d(self):
             [0.0, 0.0, 1.]]])
         assert_allclose(output, expected)
 
+        # check to ensure it handles case where a dimension == 1
+        height = 1
```

Comment:
For better testing, we shall test all edge cases of params. Something like
~~~python
@parametrize("width", (1, 5))
@parametrize("height", (1, 5))
def test_smoke(self, width, height, ...):
    ...
~~~

Reply:
So does the 3D one.

Reply:
To parameterize width and height, I'd have to reproduce the current function in the tests-- checking for 1, adding epsilon, etc. It seems off to recreate the function in the test itself

Reply:
It is better to still test based on the expected values.

I propose something like below for 2D:
```python
@parametrize("width,height,expected", [
    (1, 1, tensor_1),
    (1, 5, tensor_2),
    (5, 1, tensor_3),
    (5, 5, tensor_4),
])
def test_smoke(self, width, height, expected, ...):
```

For 3D:
```python
@parametrize("depth, width,height,expected", [
    (1, 1, 1, tensor_0),
    (1, 1, 5, tensor_1),
    (1, 5, 1, tensor_2),
    (5, 1, 1, tensor_3),
    (5, 5, 1, tensor_4),
    (5, 1, 5, tensor_5),
    (1, 5, 5, tensor_6),
    (5, 5, 5, tensor_7),
])
def test_smoke(self, depth, width, height, expected, ...):
```

Maybe nasty, but would be very sufficient. You can also go for it if you got better ideas.

Reply:
@jbohnslav please add this either as @shijianjian proposed or in a separate test. Combining two tests in the same call may cause confusion to debug in case one of them crashes